### DISCUSSION TITLE
bug: KSV028 csi is an allowed volume type

### DISCUSSION THREAD

**Discussion by joebowbeer** [aquasecurity/trivy#5768] [Bugs]
### Description

The restricted rule lists csi among the disallowed volume types:

https://github.com/aquasecurity/trivy-policies/blob/main/checks/kubernetes/pss/restricted/1_non_core_volume_types.rego#L54

But csi is listed among the allowed volume types in the referenced doc:

https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted

```
Every item in the spec.volumes[*] list must set one of the following fields to a non-null value:
  spec.volumes[*].configMap
  spec.volumes[*].csi
  spec.volumes[*].downwardAPI
  spec.volumes[*].emptyDir
  spec.volumes[*].ephemeral
  spec.volumes[*].persistentVolumeClaim
  spec.volumes[*].projected
  spec.volumes[*].secret
```

In addition, the deny message could be improved. It stipulates PersistentVolumeClaim instead of any allowed volume type.

By the way, why is this rule implemented by listing all _non-allowed_ instead of all _allowed_ types?

### Desired Behavior

Two expectations:
1. csi is an allowed volume type
2. When a volume type is denied, the suggested fix should be to provide an allowed type, not just PersistentVolumeClaim

### Actual Behavior

csi is denied

### Reproduction Steps

NOTE: This identical [issue](https://github.com/aquasecurity/defsec/issues/668) was previously reported and fixed in the defsec repo, but apparently the fix was lost during the migration to the trivy-policies repo.

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
N/A
```

### Operating System

N/A

### Version

```bash
N/A
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by AnaisUrlichs** [comment]
@simar7 I think this is relevant to you?

**Comment by nikpivkin** [comment]
@AnaisUrlichs [PR](https://github.com/aquasecurity/trivy-policies/pull/44/files) is already open for this fix 

/cc @simar7

**Comment by nikpivkin** [comment]
@simar7 `csi` was returned as a disallowed volume type in this [PR](https://github.com/aquasecurity/defsec/pull/1382/files#diff-9b66a076cf68aee9c465494bc2dc8214d03289513c9bc01f3bfb4fe859476daaR45)

**Reply by simar7** [reply]
Yes I'm aware. @chen-keinan @mjshastha could you clarify as your PR had made this change. Is this a commercial requirement?

**Reply by mjshastha** [reply]
@simar7 We incorporated "csi" into the list of disallowed volume types within the commercial regos, yet the rationale behind this inclusion eludes our recollection. You can go ahead and take "csi" off the list of disallowed volume types.

**Comment by joebowbeer** [comment]
PR https://github.com/aquasecurity/trivy-policies/pull/44 to allow `csi` volume type was merged
